web/src/components/RunActionsMenu/RunActionsMenu.tsx

```diff
@@ -52,7 +53,7 @@ const RunActionsMenu = ({resultId, testId, testSuiteId, testSuiteRunId, isRunVie
               data-cy="automate-test-button"
               key="automate-test"
               onClick={() => {
-                navigate(`/test/${testId}/run/${resultId}/automate`);
+                navigate(`/test/${testId}/run/${resultId}/automate`, {state: {origin}});
```

Comment:
What happens if the origin is empty or undefined?

Reply:
Nothing. We default it to the root page when we do the navigation.